go/signatures.go

```diff
@@ -0,0 +1,92 @@
+package ct
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+)
+
+// PublicKeyFromPEM parses a PEM formatted block and returns the public key contained within.
+func PublicKeyFromPEM(b []byte) (crypto.PublicKey, error) {
+	p, _ := pem.Decode(b)
+	if p == nil {
+		return nil, fmt.Errorf("no PEM block found in %s", string(b))
+	}
+	return x509.ParsePKIXPublicKey(p.Bytes)
+}
+
+// SignatureVerifier can verify signatures on SCTs and STHs
+type SignatureVerifier struct {
+	pubKey crypto.PublicKey
+}
+
+// NewSignatureVerifier creates a new SignatureVerifier using the passed in PublicKey.
+func NewSignatureVerifier(pk crypto.PublicKey) SignatureVerifier {
```

Comment:
Definitely (a); may as well fail early if we're only going to fail as soon as we hit the first signature.
(b) and (c) I'm torn - on the one hand technically a signature from a small RSA key or EC on the wrong curve isn't 6962 compliant, and I totally agree about putting such a hard fail into the `Signer` when it exists, on the other hand if this code is just looking at stuff from someone else's (broken) log should it flat out refuse to start?

I've done (a), and now log a stern warning for (b) and (c), but open to discussion about promoting those to fails.